examples/auth/cadc_auth_example.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+Example for authenticating with CADC TAP service
+"""
+import getpass
+
+import requests
+
+import pyvo
+import pyvo.extensions.auth.authsession
+
+# Gather login information
+data = {
+    'username': input('Username:'),
+    'password': getpass.getpass('Password:')
+}
+
+headers = {
+    'Content-Type': 'application/x-www-form-urlencoded',
+    'Accept': 'text/plain'
+}
+
+# Do the login and get the cookie
+login_url = 'https://ws-cadc.canfar.net/ac/login'
+response = requests.post(login_url, data=data, headers=headers)
+response.raise_for_status()
+cookie = '\"' + response.text + '\"'
+
+# Configure the session and run the query
+auth = pyvo.extensions.auth.authsession.AuthSession()
+auth.credentials.set_cookie('CADC_SSO', cookie)
+service = pyvo.dal.TAPService('https://www.cadc-ccda.hia-iha.nrc-cnrc.gc.ca/tap', auth)
+job = service.search('SELECT * from TAP_SCHEMA.tables')
```

Comment:
For power users I would maybe add an optional `session` of type `requests.session` attribute to the `search` or other methods in the class that hit the server. This attribute would override the internal `session(s)` created by the object to allow possible optimizations. This is optional and we should add it only if there are clear scenarios that requires it.

Reply:
The classes themselves already take the session parameter, but you're right that the some of the highest level wrappers that try to wrap everything into one line don't have this.  But someone could just make the class themselves - it's not like we're preventing something, this is just about lines of code.  I could add in the parameter, but I'm not sure anyone is calling for it.

Reply:
The classes do take the session parameter but is that session meant to be used with all the end points of the service? Passing it with the actual function allows the caller to target specific end point(s).

Reply:
Sorry what is an endpoint in this context?  Auth will only be used on URLs that call for it in the capabilities.  Otherwise you have to set the anonymous session, which will be used for all the other calls (it can still do auth, or whatever it needs to, but it is used when the negotiated security method is anonymous).